### DISCUSSION TITLE
LDAP Custom attribute map

### DISCUSSION THREAD

**Discussion by dorianim** [directus/directus#9294] [Feature Requests]
Hi,

I just installed Directus 9.0.rc100 and am very happy that it now supports LDAP, thank you so much for implementing it :D
I just need one more feature: A custom attribute map. We store the mail addresses of our users in a field other than `mail` and it would be nice if it was possible to make Directus import them from there.

Best Regards
Dorian

**Comment by dorianim** [comment]
I guess, this would have to be implemented somewhere around here: https://github.com/directus/directus/blob/ded3d4b1412a4da711dfaa31e72a4fd4e7e31acc/api/src/auth/drivers/ldap.ts#L128

**Reply by aidenfoxx** [reply]
> I just installed Directus 9.0.rc100 and am very happy that it now supports LDAP, thank you so much for implementing it :D

You're welcome!

> I just need one more feature: A custom attribute map. We store the mail addresses of our users in a field other than mail and it would be nice if it was possible to make Directus import them from there.

I will admit I was rather bias on this in the sense I used the standard Active Directory user attributes. It would probably be sufficient to expand out the LDAP query to get other standardized attributes like "surname" and "EmailAddress", and just use whichever is found. But I guess it would also be fine to add `AUTH_<provider>_EMAIL_ATTRIBUTE` etc. config options.

> I guess, this would have to be implemented somewhere around here: 

Yeah, that's the place!

**Reply by dorianim** [reply]
Ok, cool. I can try to add the `AUTH_<provider>_EMAIL_ATTRIBUTE` and do a PR, lets see :sweat_smile: ...

edit: I just created the PR: #9307
